main.go

```diff
@@ -912,9 +930,17 @@ func main() {
 
 	failCount := 0
 	syncCount := uint64(0)
+
+	v := Values{map[string]any{
+		"webhook":              webhookRunner,
+		"exechook":             exechookRunner,
+		"flHooksAsync":         *flHooksAsync,
+		"flHooksBeforeSymlink": *flHooksBeforeSymlink,
+	}}
+
 	for {
 		start := time.Now()
-		ctx, cancel := context.WithTimeout(context.Background(), *flSyncTimeout)
+		ctx, cancel := context.WithTimeout(context.WithValue(context.Background(), "hookvalues", v), *flSyncTimeout)
```

Comment:
Rather than squirreling it into ctx, how about crafting a func(hash) which you pass alongside the credentials func.  That way, all the hooks logic is defined here.

It might change the order of things like hooks vs. touch file, but I think that is OK.  May need to include the setRepoReady.  I am on mobile so this SEEMS like a good idea but I can't go try it to be  sure :)